src/react-components/room/useRoomLoadingState.js

```diff
@@ -105,35 +130,50 @@ export function useRoomLoadingState(sceneEl) {
+        if (lazyLoadMedia.current) {
+          sceneEl.addEventListener("environment-scene-loading", onEnvironmentLoading);
+          lazyLoadMediaRefValue = lazyLoadMedia.current;
+        } else {
+          console.log("Attaching room state event listeners");
+          sceneEl.addEventListener("model-loading", onObjectLoading);
+          sceneEl.addEventListener("image-loading", onObjectLoading);
+          sceneEl.addEventListener("pdf-loading", onObjectLoading);
+          sceneEl.addEventListener("video-loading", onObjectLoading);
+          sceneEl.addEventListener("model-loaded", onObjectLoaded);
+          sceneEl.addEventListener("image-loaded", onObjectLoaded);
+          sceneEl.addEventListener("pdf-loaded", onObjectLoaded);
+          sceneEl.addEventListener("video-loaded", onObjectLoaded);
+          sceneEl.addEventListener("model-error", onObjectLoaded);
+        }
         sceneEl.addEventListener("environment-scene-loaded", onEnvironmentLoaded);
         sceneEl.addEventListener("didConnectToNetworkedScene", onNetworkConnected);
       }
 
       return () => {
-        console.log("Removing room state event listeners");
-        sceneEl.removeEventListener("model-loading", onObjectLoading);
-        sceneEl.removeEventListener("image-loading", onObjectLoading);
-        sceneEl.removeEventListener("pdf-loading", onObjectLoading);
-        sceneEl.removeEventListener("model-loaded", onObjectLoaded);
-        sceneEl.removeEventListener("image-loaded", onObjectLoaded);
-        sceneEl.removeEventListener("pdf-loaded", onObjectLoaded);
-        sceneEl.removeEventListener("model-error", onObjectLoaded);
+        if (lazyLoadMediaRefValue) {
+          sceneEl.removeEventListener("environment-scene-loading", onEnvironmentLoading);
+        } else {
+          console.log("Removing room state event listeners");
+          sceneEl.removeEventListener("model-loading", onObjectLoading);
+          sceneEl.removeEventListener("image-loading", onObjectLoading);
+          sceneEl.removeEventListener("pdf-loading", onObjectLoading);
+          sceneEl.removeEventListener("video-loading", onObjectLoading);
+          sceneEl.removeEventListener("model-loaded", onObjectLoaded);
+          sceneEl.removeEventListener("image-loaded", onObjectLoaded);
+          sceneEl.removeEventListener("pdf-loaded", onObjectLoaded);
+          sceneEl.removeEventListener("video-loaded", onObjectLoaded);
+          sceneEl.removeEventListener("model-error", onObjectLoaded);
+        }
         sceneEl.removeEventListener("environment-scene-loaded", onEnvironmentLoaded);
         sceneEl.removeEventListener("didConnectToNetworkedScene", onNetworkConnected);
       };
     },
-    [sceneEl, loading, onObjectLoaded, onObjectLoading, onEnvironmentLoaded, onNetworkConnected]
+    [sceneEl, loading, onObjectLoaded, onObjectLoading, onEnvironmentLoading, onEnvironmentLoaded, onNetworkConnected]
```

Comment:
I think you need `lazyLoadMedia` in here since the effect should re-run if this changes.